rust/origen/src/generator/nodes.rs

```diff
@@ -67,24 +66,27 @@ pub enum Attrs {
     //////////////////////////////////////////////////////////////////////////////////////////////////////////
     //// JTAG nodes
     //////////////////////////////////////////////////////////////////////////////////////////////////////////
-    JTAGWriteIR(u32, BigUint, Option<BigUint>, Option<String>), // size, data, overlay_enable, overlay_str
+    JTAGWriteIR(
+        Id, // JTAG ID
+        Transaction,
+        Metadata,
+    ),
     JTAGVerifyIR(
-        u32,
-        BigUint,
-        Option<BigUint>,
-        Option<BigUint>,
-        Option<BigUint>,
-        Option<String>,
-    ), // size, data, verify_enable, capture_enable, overlay_enable, overlay_str
-    JTAGWriteDR(u32, BigUint, Option<BigUint>, Option<String>), // size, data, overlay_enable, overlay_str
+        Id, // JTAG ID
+        Transaction,
+        Metadata,
+    ),
+    JTAGWriteDR(
+        Id, // JTAG ID
+        Transaction,
```

Comment:
Do these Transaction and Metadata structures provide hooks needed to perform a read while write operation?